src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -233,8 +233,18 @@ private void evaluateSystemIndicesAccess(
     ) {
         // Perform access check is system index permissions are enabled
         boolean containsSystemIndex = requestContainsAnySystemIndices(requestedResolved);
+        boolean serviceAccountUser = user.isServiceAccount();
 
         if (isSystemIndexPermissionEnabled) {
+            if (serviceAccountUser && !containsSystemIndex) {
+                auditLog.logSecurityIndexAttempt(request, action, task);
```

Comment:
@RyanL1997 How would this PR handle the following scenario?

Extension uses its service account token to search on 2 indices: 1 regular, 1 system index. i.e.

```
GET /regular-index,.system-index/_search
{
  "query": {
    "match": {
      "foo": "bar"
    }
  }
}
```

Its my understanding in the design from @scrawfor99 and @peternied that a service account would be conventionally tied to a role that is limited to only system indices and may look something like:

```
ext1role:
  description: "Extension1 Role"
  index_permissions:
    - index_patterns:
        - '.ext1-*'
        - '.system-index'
      allowed_actions:
        - "system:admin/system_index"
```

but what if the role was defined with regular indexes like:

```
ext1role:
  description: "Extension1 Role"
  index_permissions:
    - index_patterns:
        - '.ext1-*'
        - '.system-index'
        - 'regular-index*'
      allowed_actions:
        - "system:admin/system_index"
        - "read"
```

In the case of the second role would it be possible for the account to search on indices matching `regular-index*`?

Reply:
> In the case of the second role would it be possible for the account to search on indices matching regular-index*?

Hi @cwperks , thanks for the detailed feedback. I think for the above question, the answer is no. As long as this account is a service account, it wont be able to access to non-system index for this implementation.

Yesterday I did a manual testing with @DarshitChanpura yesterday, and the procedure was captured in this [issue in my fork](https://github.com/RyanL1997/security/issues/2). I think the procedure of what we did also covered this case since the role we were using for testing is having a `*` wild card index permission.

Reply:
Actually you were correct @cwperks, and it didnt handle the case you provided. I was only check if the request `Boolean containSystemIndex`, so that if it is a mixed the request with both non-system index and system-index it will still be considered as true for the above condition. According to that, I have pushed another [commit](https://github.com/opensearch-project/security/pull/3607/commits/ac9416ee018c75b986f87233608055d98d3b12c9) to handle this 'mixed' case. So now, if there is a mixed request, it will filter out the regular index from the request and then only do the system index only. Thanks for pointing this out btw. I also added a test case for this, and the detailed parsing to prove it only do the system index request will be added later.